### PR TITLE
creating a data model

### PR DESCRIPTION
**Pull request by Lucas-Erkana**
- [ ] Be sure to reference the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) from the sneak peek if you have any questions about table or column names or data relationships.
- [ ]  Create a new app. Give it a meaningful name.
- [ ]  Initialize your project with Git.
- [ ]  Make sure that your project has Postgres database set up.
- [ ]  Build your project schema.
- [ ]  Create and run the necessary migration files.
- [ ]  Table and column names should match the ERD diagram.
- [ ]  note: photo for users table should be a link to an image
- [ ]  Foreign keys should be included.
- [ ]  Pay attention that in posts table the column holding the foreign key to the users table should be named author_id
- [ ]  All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/migrate/20230613175409_add_user_ref_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
+  end
+end
```
Comment (GzimAsani) [inline db/migrate/20230613175409_add_user_ref_to_posts.rb]: - [ ] According to the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) diagram the :user table should be refered as author, kindly add it like this 

```ruby
 t.references :author, foreign_key: { to_table: 'users' }
```

File: db/migrate/20230613175402_add_user_ref_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
+  end
+end
```
Comment (GzimAsani) [inline db/migrate/20230613175402_add_user_ref_to_comments.rb]: - [ ] According to the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) diagram the :user table should be refered as author, kindly add it like this 

```ruby
 t.references :author, foreign_key: { to_table: 'users' }
```

File: db/migrate/20230613175604_add_post_ref_to_likes.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddPostRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :post, null: false, foreign_key: true
+  end
+end
```
Comment (GzimAsani) [inline db/migrate/20230613175604_add_post_ref_to_likes.rb]: - [ ] According to the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) diagram the :user table should be refered as author, kindly add it like this 

```ruby
 t.references :author, foreign_key: { to_table: 'users' }
```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @Lucash2022                  :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with using PostgreSQL database:+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230613175604_add_post_ref_to_likes.rb (hunk 3 above)
Comment (Lucas-Erkana) [reply db/migrate/20230613175604_add_post_ref_to_likes.rb]: Thank you for the review. I have made the change at this [commit](https://github.com/Lucash2022/Blog_App/pull/2/commits/9680b1a754e5843ea70ddcef6ed3a66c95103e1d#)

File: db/migrate/20230613175402_add_user_ref_to_comments.rb (hunk 2 above)
Comment (Lucas-Erkana) [reply db/migrate/20230613175402_add_user_ref_to_comments.rb]: Thank you for the review. I have made the change at this [commit](https://github.com/Lucash2022/Blog_App/pull/2/commits/9680b1a754e5843ea70ddcef6ed3a66c95103e1d#)

File: db/migrate/20230613175409_add_user_ref_to_posts.rb (hunk 1 above)
Comment (Lucas-Erkana) [reply db/migrate/20230613175409_add_user_ref_to_posts.rb]: Thank you for the review. I have made the change at this [commit](https://github.com/Lucash2022/Blog_App/pull/2/commits/9680b1a754e5843ea70ddcef6ed3a66c95103e1d#)

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Lucash2022 

Your project is complete! Great job for implementing the changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- ERD diagram followed
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**